services/ingestion/etl/cordis/csv/src/lib/transform.js

```diff
@@ -0,0 +1,265 @@
+            nuts: [],
+            address: '',
+            postal_code: '',
+            town: '',
+            centroid: null,
+            location: null,
+          };
+        }
+
+        // empty string of nothing else
+        return code;
+      })
+      // Remove empty
+      .filter(country => country);
+
+    locations.push(...loc);
+  }
+
+  return locations;
+};
+
+/**
+ * Preprocess third parties
+ * Input fields taken from the `record` are:
+ * - `coordinator`
+ * - `coordinatorCountry`
+ * - `participants`
+ * - `participantCountries`
+ *
+ * @memberof CordisCsvTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {Array} List of {ThirdParty} objects
+ */
+const getThirdParties = record => {
+  const thirdParties = [];
+
+  if (record.coordinator) {
+    thirdParties.push({
+      name: record.coordinator,
+      type: '',
+      address: '',
+      region: '',
+      role: 'coordinator',
+      country: record.coordinatorCountry || '',
+      website: '',
+      phone: '',
+      email: '',
+    });
+  }
+
+  if (record.participants) {
+    const participants = record.participants
+      .split(';')
+      .filter(participant => participant)
+      .map(participant => ({
+        name: participant.trim(),
+        type: '',
+        address: '',
+        region: '',
+        role: 'participant',
+        website: '',
+        phone: '',
+        email: '',
+      }));
+
+    record.participantCountries
+      .split(';')
+      .filter(participantCountry => participantCountry)
+      .forEach((country, participantKey) => {
+        if (participants[participantKey]) {
+          const countryCode = country.trim().toUpperCase();
+          participants[participantKey].country = getCountryCode(countryCode);
+        }
+      });
+
+    thirdParties.push(...participants);
+  }
+
+  return thirdParties;
+};
+
+/**
+ * Format date
+ *
+ * @memberof CordisCsvTransform
+ * @param {Date} date Date in `YYYY-MM-DD` (ISO) format
+ * @returns {Date} The date formatted into an ISO 8601 date format
+ *
+ * @example
+ * input => "2018-12-31"
+ * output => "2018-12-31T00:00:00.000Z"
+ */
+const formatDate = date => {
+  if (!date || typeof date !== 'string') return null;
+
+  try {
+    return new Date(date).toISOString();
+  } catch (e) {
+    return null;
+  }
+};
+
+/**
+ * Map fields for CORDIS producer, CSV file types
+ *
+ * Example input data: {@link https://github.com/ec-europa/eubfr-data-lake/blob/master/services/ingestion/etl/cordis/csv/test/stubs/record.json|stub}
+ *
+ * Transform function: {@link https://github.com/ec-europa/eubfr-data-lake/blob/master/services/ingestion/etl/cordis/csv/src/lib/transform.js|implementation details}
+ * @name CordisCsvTransform
+ * @param {Object} record Piece of data to transform before going to harmonized storage.
+ * @returns {Project} JSON matching the type fields.
+ */
+export default (record: Object): Project => {
+  // Preprocess third parties
```

Comment:
If `record` is not defined or is null, the function will crash. Can you add a simple check here? `if (!record) return something;`

Reply:
Added the same check to all other transform functions.